### ISSUE TITLE
unresolved import `core::unicode`

### ISSUE THREAD

**Issue by wangjiawen2013** [evcxr/evcxr#365]
Hi,
I cannot activate the strings feature. I am using evcxr.
:dep polars = {version = "0.35.4", features = ["strings","ndarray","lazy","concat_str","rows","round_series"]}
The following error occured:
unresolved import `core::unicode`

Then I abandoned evcxr and vscode and switch to use polars in main.rs . This time no error occurred. It seems that it is caused by evcxr.

And this is an releated issue: https://github.com/pola-rs/polars/issues/13056

**Comment by davidlattimore** [comment]
I just attempted to reproduce by opening evcxr, then running the `:dep` command you gave. It completed without error. Could there be something else that's needed in order to reproduce?

`core::unicode` is an unstable API, so would only be available on the nightly versions of rustc. Perhaps you're running nightly, but it's an older version? Could you be using a different version of rustc when compiling code outside of evcxr than inside?

**Comment by wangjiawen2013** [comment]
I'll check it. Could you show your evcxr version and rustc version ?

**Comment by davidlattimore** [comment]
Sure:

```
rustc --version
rustc 1.76.0 (07dca489a 2024-02-04)

evcxr --version
evcxr_repl 0.17.0
```

For good measure, and since I noticed that polars has the `version_check` crate in its dependencies, I also ran the following from evcxr:

```
:dep version_check = "0.9.4"
version_check::triple()
```

Which reported the following:

```
Some((Version(4299948032, (1, 76, 0)), Channel(Stable), Date(1036356)))
```

**Comment by wangjiawen2013** [comment]
Hi, now I cannot connect my vscode to rust kernel, but I can connect to python kernel

![image](https://github.com/evcxr/evcxr/assets/29703450/c5c231f4-430d-48a4-b8e8-cdd406e53340)

**Comment by davidlattimore** [comment]
Does it work if you run `jupyer notebook` then connect to that from your web browser?

**Comment by wangjiawen2013** [comment]
It doesn't work. The browser is empty.

![image](https://github.com/evcxr/evcxr/assets/29703450/f3e4dba0-e9f7-4980-983c-3a061d08b083)

**Comment by wangjiawen2013** [comment]
(py311) [wangjw@localhost work]$ jupyter notebook
[I 2024-03-12 13:35:24.794 ServerApp] jupyter_lsp | extension was successfully linked.
[W 2024-03-12 13:35:24.795 TerminalsExtensionApp] Config option `kernel_spec_manager_class` not recognized by `TerminalsExtensionApp`.  Did you mean `terminal_manager_class`?
[W 2024-03-12 13:35:24.796 TerminalsExtensionApp] Config option `kernel_spec_manager_class` not recognized by `TerminalsExtensionApp`.  Did you mean `terminal_manager_class`?
[W 2024-03-12 13:35:24.797 TerminalsExtensionApp] Config option `kernel_spec_manager_class` not recognized by `TerminalsExtensionApp`.  Did you mean `terminal_manager_class`?
[I 2024-03-12 13:35:24.798 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-03-12 13:35:24.799 LabApp] Config option `kernel_spec_manager_class` not recognized by `LabApp`.
[W 2024-03-12 13:35:24.800 LabApp] Config option `kernel_spec_manager_class` not recognized by `LabApp`.
[W 2024-03-12 13:35:24.802 LabApp] Config option `kernel_spec_manager_class` not recognized by `LabApp`.
[I 2024-03-12 13:35:24.802 ServerApp] jupyterlab | extension was successfully linked.
[W 2024-03-12 13:35:24.803 JupyterNotebookApp] Config option `kernel_spec_manager_class` not recognized by `JupyterNotebookApp`.
[W 2024-03-12 13:35:24.804 JupyterNotebookApp] Config option `kernel_spec_manager_class` not recognized by `JupyterNotebookApp`.
[W 2024-03-12 13:35:24.806 JupyterNotebookApp] Config option `kernel_spec_manager_class` not recognized by `JupyterNotebookApp`.
[I 2024-03-12 13:35:24.806 ServerApp] notebook | extension was successfully linked.
[I 2024-03-12 13:35:24.953 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-03-12 13:35:25.502 ServerApp] [nb_conda_kernels] enabled, 2 kernels found
[I 2024-03-12 13:35:25.518 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-03-12 13:35:25.519 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-03-12 13:35:25.520 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-03-12 13:35:25.521 LabApp] JupyterLab extension loaded from /home/wangjw/programs/miniconda3/envs/py311/lib/python3.11/site-packages/jupyterlab
[I 2024-03-12 13:35:25.521 LabApp] JupyterLab application directory is /home/wangjw/programs/miniconda3/envs/py311/share/jupyter/lab
[I 2024-03-12 13:35:25.521 LabApp] Extension Manager is 'pypi'.
[I 2024-03-12 13:35:25.531 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-03-12 13:35:25.533 ServerApp] notebook | extension was successfully loaded.
[I 2024-03-12 13:35:25.533 ServerApp] The port 8888 is already in use, trying another port.
[I 2024-03-12 13:35:25.533 ServerApp] The port 8889 is already in use, trying another port.
[I 2024-03-12 13:35:25.533 ServerApp] The port 8890 is already in use, trying another port.
[I 2024-03-12 13:35:25.534 ServerApp] Serving notebooks from local directory: /home/wangjw/data/work
[I 2024-03-12 13:35:25.534 ServerApp] Jupyter Server 2.13.0 is running at:
[I 2024-03-12 13:35:25.534 ServerApp] http://localhost:8891/tree?token=c3c8ef483a4957532ebea67cb0f1d2ee79ef99f9cf042187
[I 2024-03-12 13:35:25.534 ServerApp]     http://127.0.0.1:8891/tree?token=c3c8ef483a4957532ebea67cb0f1d2ee79ef99f9cf042187
[I 2024-03-12 13:35:25.534 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 2024-03-12 13:35:32.932 ServerApp] 
    
    To access the server, open this file in a browser:
        file:///home/wangjw/.local/share/jupyter/runtime/jpserver-128015-open.html
    Or copy and paste one of these URLs:
        http://localhost:8891/tree?token=c3c8ef483a4957532ebea67cb0f1d2ee79ef99f9cf042187
        http://127.0.0.1:8891/tree?token=c3c8ef483a4957532ebea67cb0f1d2ee79ef99f9cf042187
[I 2024-03-12 13:35:33.556 ServerApp] Skipped non-installed server(s): bash-language-server, dockerfile-language-server-nodejs, javascript-typescript-langserver, jedi-language-server, julia-language-server, pyright, python-language-server, python-lsp-server, r-languageserver, sql-language-server, texlab, typescript-language-server, unified-language-server, vscode-css-languageserver-bin, vscode-html-languageserver-bin, vscode-json-languageserver-bin, yaml-language-server
START /usr/bin/firefox "/home/wangjw/.local/share/jupyter/runtime/jpserver-128015-open.html"

**Comment by davidlattimore** [comment]
If the page is blank, then I'd say something is wrong with the Jupyter installation.

**Comment by wangjiawen2013** [comment]
The page is blank because of too many tasks were running on our servers.  I finally find that this is a platform-specific problem becuase everything goes well on my windows11 computer, but it's a messy on my centos server.

After lots of debugging, I find the key is that we must keep rustc toochain and rust-analyzer version matching for evcxr_repl/evcxr_jupyter to work.  On my server, I installed `rustc 1.78.0-nightly` , `rust-analyzer 1.78.0-nightly` and `evcxr_repl 0.17.0`, this combination solved all the questions I met, including "unresolved import core::unicode" and https://github.com/evcxr/evcxr/issues/328

By the way, I installed the **matching** rust-analyer using `rustup component add rust-analyzer`
